coordinator/internal/verify/verify.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2025 Edgeless Systems GmbH
+// SPDX-License-Identifier: BUSL-1.1
+
+package verify
+
+import (
+	"context"
+	"crypto/sha512"
+	"encoding/json"
+	"errors"
+	"io"
+	"log"
+	"net/http"
+
+	"github.com/edgelesssys/contrast/coordinator/internal/stateguard"
+	"github.com/edgelesssys/contrast/coordinator/internal/userapi"
+	"github.com/edgelesssys/contrast/internal/atls"
+	"github.com/edgelesssys/contrast/sdk"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// Request is the wire-format for incoming /verify requests.
+// The nonce is expected to be base64-encoded.
+type Request struct {
+	Nonce []byte `json:"nonce"`
+}
+
+// Handler implements http.Handler for POST /verify.
+type Handler struct {
+	Issuer     atls.Issuer
+	StateGuard *stateguard.Guard
+}
+
+func (h *Handler) getResponse(ctx context.Context, req *Request) (*sdk.CoordinatorState, error) {
+	// state knows the lates transition
+	state, err := h.StateGuard.GetState(ctx)
+	switch {
+	case errors.Is(err, stateguard.ErrNoState):
+		return nil, status.Error(codes.FailedPrecondition, userapi.ErrNoManifest.Error())
```

Comment:
Instead of the `status.Error`s, which are for gRPC, we should work with HTTP error codes here.